### PR TITLE
config rollup bundle

### PR DESCRIPTION
**Pull request by YehudaGold**
this pull request is based on  #1.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     // "noEmitHelpers": true,                            /* Disable generating custom helper functions like `__extends` in compiled output. */
     // "noEmitOnError": true,                            /* Disable emitting files if any type checking errors are reported. */
     // "preserveConstEnums": true,                       /* Disable erasing `const enum` declarations in generated code. */
-    // "declarationDir": "./",                           /* Specify the output directory for generated declaration files. */
+    "declarationDir": ".",                           /* Specify the output directory for generated declaration files. */
```
Comment (shachar731945) [inline tsconfig.json]: why would you want dec files in the root and not on the dist

File: tsconfig.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
 
     /* Emit */
     "declaration": true,                                 /* Generate .d.ts files from TypeScript and JavaScript files in your project. */
-    "declarationMap": true,                              /* Create sourcemaps for d.ts files. */
```
Comment (shachar731945) [inline tsconfig.json:46]: don't you want to add the declerationMap for the transpile:sourcemap for debugging the code?

File: .npmignore [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 .idea/
 .vscode/
 
 # development
 src/
-test/
+tests/
 
-# Ignore map sources
-*.js.map
```
Comment (shachar731945) [inline .npmignore]: Would you like to upload the .js.map files to npm?

File: package.json [hunk 4]
```diff
@@ -2,25 +2,35 @@
...
+  "author": "shahar avni",
+  "contributors": [
+    "yehuda goldberg"
+  ],
+  "license": "MIT",
+  "main": "dist/bundle.cjs.js",
+  "module": "dist/bundle.esm.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "test": "npx jest",
```
Comment (shachar731945) [inline package.json]: Didn't you want in the previous PR #1 to make a script with .d.ts and .js.map and the other one without?

File: tsconfig.json (hunk 1 above)
Comment (YehudaGold) [reply tsconfig.json]: `declarationDir` is relative to` outDir`, you can see the output in nested folder 'dist/dist' if you don`t correct it.

https://github.com/rollup/plugins/tree/master/packages/typescript/#declaration-output-with-outputfile

File: tsconfig.json (hunk 2 above)
Comment (YehudaGold) [reply tsconfig.json:46]: the rollup flag override this setting

File: .npmignore (hunk 3 above)
Comment (YehudaGold) [reply .npmignore]: there some benefits to include map files for debugging, but wee decided to not generate the  map files in publish transpile so there isn`t files to ignore

Comment (YehudaGold) [reply .npmignore]: I can revert this if you want

File: package.json (hunk 4 above)
Comment (YehudaGold) [reply package.json]: look in the file, i added `transpile` and `transpile:sourcemap` scripts.
The script `build` for publishing the package run `clean` and `transpile`.

**Pull request merged**